content/blog/2022/2022-09-01-schedule-cron-job-in-node.md

```diff
@@ -0,0 +1,319 @@
+
+To create our email scheduler application Insert the following code into the `index.js` file:
+```javascript
+const express = require("express");
+const cron = require("node-cron");
+const nodemailer = require("nodemailer");
+app = express();
+
+//send email after 1 minute
+cron.schedule("1 * * * *", function () {
+  mailService();
+});
+
+function mailService() {
+  let mailTransporter = nodemailer.createTransport({
+    service: "gmail",
+    auth: {
+      user: "<your-email>@gmail.com",
+// use generated app password for gmail
+      pass: "***********",
+    },
+  });
+
+  // setting credentials
+  let mailDetails = {
+    from: "<your-email>@gmail.com",
+    to: "<user-email>@gmail.com",
+    subject: "Test Mail using Cron Job",
+    text: "Node.js Cron Job Email Demo Test from Reflectoring Blog",
+  };
+
+  // sending email
+  mailTransporter.sendMail(mailDetails, function (err, data) {
+    if (err) {
+      console.log("error occured", err.message);
+    } else {
+      console.log("---------------------");
+      console.log("email sent successfully");
+    }
+  });
+}
+
+app.listen(3000, () => {
+  console.log("application listening.....");
+});
+
+```
+In the above code we are using the node-mailer and node-cron dependencies we installed earlier. The `node-mailer` dependency gives us the feature of sending e-mails from our Node.js applications, using any email service of your choice.
+
+We scheduled our mail to be sent at `At 1 minute` interval using Gmail.
+
+Run the script `node index.js`
+
+`output:`
+```
+application listening.....
+---------------------
+email sent successfully
+```
+Check your inbox to confirm the email is sent.
+
+### 3. Monitoring Server Resources Over Time
+Cron jobs can be used to schedule logging tasks and monitor server resources in our node.js applications. Let's say something happened, like a network delay or a warning message. We can schedule a cron job to log at a specific time or interval to track our server status, this can act as an automatic monitor over time.
+
+In this section, we will log the application's server resources in `csv` format, this makes our logged data more machine-readable. The generated `.csv` file can be imported into a spreadsheet to create graphs for more advanced use cases.
+
+Insert the following code into the `index.js` file to generate the `.csv` file at the scheduled time:
+```javascript
+const process = require("process");
+const fs = require("fs");
+const os = require("os");
+const cron = require("node-cron");
+const express = require("express");
+
+app = express();
+
+// setting a cron job for every 15 seconds
+cron.schedule("*/15 * * * * *", function () {
+  let heap = process.memoryUsage().heapUsed / 1000000;
+  let data = new Date().toISOString();
+  const freeMemory = Math.round((os.freemem() * 100) / os.totalmem()) + "%";
+
+  //                 date | heap used | free memory
+  let csv = `${dataa}, ${heap}, ${freeMemory}\n`;
```

Comment:
This can't work, can it? The variable `${dataa}` doesn't exist. (also, the variable name shouldn't be `data`, but `date`.

I couldn't find a pull request to the code-examples repo, can you please open one again (I think you closed the previous ones).